spring-grpc-core/src/main/java/org/springframework/grpc/server/ShadedNettyGrpcServerFactory.java

```diff
@@ -55,7 +55,7 @@ protected NettyServerBuilder newServerBuilder() {
 				.bossEventLoopGroup(new EpollEventLoopGroup(1))
 				.workerEventLoopGroup(new EpollEventLoopGroup());
 		}
-		return super.newServerBuilder();
+		return NettyServerBuilder.forAddress(socketAddress(), credentials());
```

Comment:
I think this replacement makes sense and is "safe" as the previous code called:
```java
Grpc.newServerBuilderForPort(port(), credentials());
```
which ultimately called:
```java
ProtocolNegotiators.FromServerCredentialsResult result = ProtocolNegotiators.from(creds);
if (result.error != null) {
    return NewServerBuilderResult.error(result.error);
}
return NewServerBuilderResult.serverBuilder(
        new NettyServerBuilder(new InetSocketAddress(port), result.negotiator));
```
and the new code `NettyServerBuilder.forAddress(socketAddress(), credentials());` does the same effectively but passes in the hostname not just the port. 

Previously we were simply disregarding any host info (as noted in your issue).